### ISSUE TITLE
Rootless feature of OS X 10.11 ElCapitain does not allow to install in /usr/bin

### ISSUE THREAD

**Issue by Vietwoojagig** [rbsec/sslscan#68]
Hi,
I'm using this script to compile and install sslscan on OS X:
```
#!/bin/bash

TEMPDIR=~/.compilesslscan

rm -Rf $TEMPDIR

mkdir $TEMPDIR
cd $TEMPDIR

git clone https://github.com/rbsec/sslscan.git

cd sslscan

make static
sudo make -B install

rm -Rf $TEMPDIR
sslscan --version
```
Everything works fine, except
sudo make -B install
fails:
cp sslscan /usr/bin/
cp: /usr/bin/sslscan: Operation not permitted
In OS X 10.11 it is is not allowed to to change the content of /user/bin.
But from what I see, /usr/local/bin is possible and it is in the PATH environment.

**Comment by connection** [comment]
If you change the following line in the Makefile it should work:

`BINPATH   = $(DESTDIR)/usr/bin/`

Change to 

`BINPATH   = $(DESTDIR)/usr/local/bin/`

**Comment by apjanke** [comment]
Perhaps `/usr/local` should be the default installation location, instead of `/usr`? That's more of a conventional location for software that is built and installed by the users, as opposed to that provided by the vendor or packager of the OS.

**Comment by Vietwoojagig** [comment]
I changed in the Makefile

`BINDIR    = $(PREFIX)/bin`
to
`BINDIR    = $(PREFIX)/local/bin`

`sudo make -B install
`
seems to require an existing version of openssl in /usr/local/ssl.
If there is no openssl version in /usr/local/ssl the output is this
```
cc -o sslscan -Wall -Wformat=2 -L/usr/local/ssl/lib/ -L/usr/local/opt/openssl/lib -I/usr/local/ssl/include/ -I/usr/local/ssl/include/openssl/ -I/usr/local/opt/openssl/include  -DVERSION=\"1.11.4-rbsec-wip\" sslscan.c -lssl -lcrypto
ld: warning: directory not found for option '-L/usr/local/ssl/lib/'
```
If there is an existing version of openssl, and this version was not configured with ./Configure darwin64-x86_64-cc, other warnings will be created.

BTW: during the compilation of openssl during make static, at one moment in time, there is the following output:

```
Configured for darwin64-x86_64-cc.

*** Because of configuration changes, you MUST do the following before
*** building:

	make depend
/Library/Developer/CommandLineTools/usr/bin/make -C openssl depend
making depend in crypto...
making depend in crypto/objects...
```

And finally:
after compiling everything:
```
$ sslscan --version
		1.11.4-rbsec-wip
		OpenSSL 1.0.2g  1 Mar 2016
		OpenSSL version does not support SSLv2
		SSLv2 ciphers will not be detected
```

**Comment by apjanke** [comment]
You shouldn't need to alter the Makefile to install under `/usr/local` any more; it now has PREFIX support (since 3ed8acf3ba3da9c9dd47a195bc004de26a36234d). Just pass the PREFIX` argument to `make`:

```
make install PREFIX=/usr/local
```

BTW, it looks like that `make install` is rebuilding a dynamically-linked `sslscan` and blowing away the result of the previous `make static` because `install` depends on `sslscan` and [the `sslscan` target is marked as phony](https://github.com/rbsec/sslscan/blob/4c6c17adc7b78b462883147a2dd8a9363e90531a/Makefile#L41).

**Comment by rbsec** [comment]
@apjanke good spot on the install bug - I almost never use `make install` so it's not one I've ever come across. Should be fixed in 0d2cd58. Doing it in bash seems like a bit of an ugly fix, but I couldn't see a nice way to do the check natively in the Makefile (although someone who knows a bit more about writing Makefiles may be able to find a cleaner solution).

@Vietwoojagig as mentioned above, you can use PREFIX to change the install directory for sslscan if your OS won't let you write to /usr/bin/
